--- uae-0.8.15/configure.in.deush	Fri Jan 12 08:46:09 2001
+++ uae-0.8.15/configure.in	Fri Jan 12 08:55:24 2001
@@ -538,7 +538,7 @@
 
 if test $ac_cv_prog_gcc = yes; then
   WARNING_CFLAGS="-Wall -Wno-unused -Wno-format -W -Wmissing-prototypes -Wstrict-prototypes"
-  OPTIMIZE_CFLAGS="-O2 -fomit-frame-pointer"
+  OPTIMIZE_CFLAGS="-fomit-frame-pointer"
   dnl Not nice, but needed to turn off -g
   CFLAGS=
 fi
@@ -727,7 +727,7 @@
   else if [[ "x$CPU" = "xi386" ]]; then
      dnl Sam, I don't think we can get here.
      MACHDEP=md-i386-gcc
-     CFLAGS="$CFLAGS -DX86_ASSEMBLY -fno-strength-reduce -DREGPARAM=\"__attribute__((regparm(3)))\""
+     CFLAGS="$CFLAGS -DX86_ASSEMBLY -DREGPARAM=\"__attribute__((regparm(3)))\""
   else
      echo "Unsupported CPU: $CPU!"
      exit 5
