--- uae-0.8.25-20040302/src/install_libscg.orig	2004-03-03 04:33:10.000000000 +0100
+++ uae-0.8.25-20040302/src/install_libscg	2004-04-18 14:13:41.391235234 +0200
@@ -44,11 +44,18 @@
 #
 uaedir=`dirname "$0"`
 
-cp -R -f "$cdrecinc/btorder.h" "$cdrecinc/intcvt.h" "$cdrecinc/mconfig.h" \
-         "$cdrecinc/prototyp.h" "$cdrecinc/utypes.h" "$cdrecorddir/libscg/scg" \
-	 "$cdrecorddir/incs/$machine/xconfig.h" "$uaedir/include"
+cp -R -f -L "$cdrecinc/btorder.h" \
+	"$cdrecinc/intcvt.h" \
+	"$cdrecinc/mconfig.h" \
+	"$cdrecinc/prototyp.h" \
+	"$cdrecinc/utypes.h" \
+	"$cdrecinc/scg" \
+	"$cdrecorddir/incs/$machine/xconfig.h" \
+	"$uaedir"/include
+
 cp -f "$cdrecorddir/libs/$machine/libscg.a" \
-      "$cdrecorddir/libs/$machine/libschily.a" "$uaedir"
+      "$cdrecorddir/libs/$machine/libschily.a" \
+      "$cdrecorddir/libs/$machine/libdeflt.a" "$uaedir"
 
 case $machine in
   *-darwin-*)
